theano/tensor/basic.py

```diff
@@ -4219,6 +4219,12 @@ def roll(x, shift, axis=None):
         else:
             axis = 0
 
+    if axis < 0:
```

Comment:
Can you add a test in the file tensor/tests/test_basic.py to make sure this don't get broken in the futur?